app/models/visit.rb

```diff
@@ -84,6 +89,15 @@ class Visit < ActiveRecord::Base
     end
   end
 
+  def book_to_nomis_opt_out
+    # TODO: Changes in Rails 5 to `ActiveRecord::Type::Boolean.new.cast(string)`
+    ActiveRecord::Type::Boolean.new.type_cast_from_database(@book_to_nomis_opt_out)
```

Comment:
Yep, setter makes more sense.

Lets take a look together on other ways this could be done as you seem to have other ideas.